### PR TITLE
all: add force option for metric system

### LINKED ISSUE #17601: eth.hashrate always returns 0
Hi there,

please note that this is an issue tracker reserved for bug reports and feature requests.

For general questions please use the gitter channel or the Ethereum stack exchange at https://ethereum.stackexchange.com.

#### System information

Geth version: 1.8.15-stable
OS & Version: Linux (Ubuntu 18.04 64-bit)
Commit hash : (if `develop`)

#### Expected behaviour
eth.hashrate returns some non-zero value

#### Actual behaviour
eth.hashrate always returns zero on my Intel Core i7-7500U laptop (512G SSD PCIe)

$ geth --testnet --port 30303 --mine --minerthreads=4 --cache=2048

> net.peerCount
22
> eth.hashrate
0
> eth.getBlock("latest").difficulty
6042033910
> eth.blockNumber
3990598
> eth.getBalance(eth.coinbase)
0

#### Steps to reproduce the behaviour
miner.stop()
miner.start(4)
eth.hashrate

while geth has been eating 95% CPU for hours

#### Backtrace
N.A.

#### Note
Found this StackOverflow question https://ethereum.stackexchange.com/questions/51220/cpu-mining-on-testnet-using-geth-on-ubuntu-eth-hashrate-0 but it doesn't really help much

### PR DESCRIPTION
**Pull request by rjl493456442**
fix #17601

### CODE REVIEW

**Comment by nonsense** [comment]
@rjl493456442 could you please add a short summary on why this is necessary?

I understand that this is needed because you want to record `hashrate` even if the Metrics system is not enabled - why is this necessary?

I am sure this has been discussed already, but please help me understand as well :)

**Comment by nonsense** [comment]
@rjl493456442 sure, it makes sense. I don't have strong feelings about it, so if you find this helpful, I approve.

**Comment by karalabe** [comment]
Would't it make more sense to add a new set of methods instead?

E.g. Instead of `func NewRegisteredCounter(name string, r Registry, force bool) Counter` we would have the original `func NewRegisteredCounter(name string, r Registry) Counter` and an additional new `func NewRegisteredCounterForced(name string, r Registry) Counter`?

This way we would retain the original API, would not introduce a big refactor all over the codebase and future metrics would remain clean as they are now. We'd only need to explicitly tag a couple metrics that are always needed.

**Comment by rjl493456442** [comment]
@karalabe I just explicitly tag the `meter` in order to achieve a minimum range of modifications.

There are many different types of metrics(`Counter`, `Timer`,`Gauge`, etc), and more complicated is some metrics relays on other metrics. I don't want to make the metrics too complicated so only `meter` is tagged.

File: ethdb/database.go [hunk 1]
```diff
@@ -155,15 +155,12 @@ func (db *LDBDatabase) LDB() *leveldb.DB {
...
-	}
-	// Initialize write delay metrics no matter we are in metric mode or not.
+	// Initialize all the metrics collector at the requested prefix
+	db.compTimeMeter = metrics.NewRegisteredMeter(prefix+"compact/time", nil)
+	db.compReadMeter = metrics.NewRegisteredMeter(prefix+"compact/input", nil)
+	db.compWriteMeter = metrics.NewRegisteredMeter(prefix+"compact/output", nil)
+	db.diskReadMeter = metrics.NewRegisteredMeter(prefix+"disk/read", nil)
+	db.diskWriteMeter = metrics.NewRegisteredMeter(prefix+"disk/write", nil)
 	db.writeDelayMeter = metrics.NewRegisteredMeter(prefix+"compact/writedelay/duration", nil)
 	db.writeDelayNMeter = metrics.NewRegisteredMeter(prefix+"compact/writedelay/counter", nil)
```
Comment (karalabe) [inline ethdb/database.go]: Don't we need Forced here?

**Pull request merged**